stdlib/src/builtin/value.mojo

```diff
@@ -145,3 +145,21 @@ trait RepresentableCollectionElement(CollectionElement, Representable):
     """
 
     pass
+
+
+trait Indexer:
```

Comment:
In Python dunder methods are not supposed to be called directly. Instead, we use a method that will call the dunder method for us (and possibly do extra logic).
For example `int()` calls `__int__`, `*` calls `__mul__`, `.` calls `__getattr__`, etc...

In the case of `__index__`, it's `operator.index()`. Since we don't have an `operator` module yet, maybe let's add the `index` function in the builtins for now? I don't think we will want to convert all the `x.__index__()` to `index(x)` later on, let's do it now in this PR.

Reply:
Oh silly me, I originally didn't do that since I saw we had a bunch of free functions already in `utils/index.mojo` named index, but those start with a capital letter so there's no risk of it clashing. I'll fix this!